# Exploratory Data Analysis – Online Retail Dataset

## 🎯 Business Objective
The objective of this project is to analyze retail transaction data to understand customer purchasing behavior, revenue trends, and product demand patterns.

## 🏢 Business Context
This dataset contains transactional data from a UK-based online retail store. Each row represents a product purchased within a transaction.

The dataset includes:
- Transaction details
- Product information
- Customer details
- Purchase timestamps

## ❓ Business Questions
- Which country generates the highest revenue?
- What products sell the most?
- Are there seasonal sales patterns?
- Are there abnormal transactions or outliers?
- What factors influence revenue?

## 📊 Expected Impact
The analysis will help understand sales performance, customer behavior, and revenue drivers.


In [2]:
import pandas as pd
import numpy as np

In [4]:
df= pd.read_csv(r"C:\Users\niyas\python\EDA Project\E-Commerce Analysis - UK\data\raw\data.csv")
df.head()
    

InvoiceNo StockCode                          Description  Quantity  \
0    536365    85123A   WHITE HANGING HEART T-LIGHT HOLDER         6   
1    536365     71053                  WHITE METAL LANTERN         6   
2    536365    84406B       CREAM CUPID HEARTS COAT HANGER         8   
3    536365    84029G  KNITTED UNION FLAG HOT WATER BOTTLE         6   
4    536365    84029E       RED WOOLLY HOTTIE WHITE HEART.         6   

      InvoiceDate  UnitPrice  CustomerID         Country  
0  12/1/2010 8:26       2.55     17850.0  United Kingdom  
1  12/1/2010 8:26       3.39     17850.0  United Kingdom  
2  12/1/2010 8:26       2.75     17850.0  United Kingdom  
3  12/1/2010 8:26       3.39     17850.0  United Kingdom  
4  12/1/2010 8:26       3.39     17850.0  United Kingdom

## Dataset Preview Observations

- Dataset successfully loaded.
- Dataset contains transaction-level retail data.
- Each row represents a product purchased in a transaction.


In [5]:
df.shape

(541909, 8)

## Dataset Shape

The dataset contains 541909 rows and 8 columns.


In [6]:
df.columns

Index(['InvoiceNo', 'StockCode', 'Description', 'Quantity', 'InvoiceDate',
       'UnitPrice', 'CustomerID', 'Country'],
      dtype='object')